### ISSUE TITLE
ComposeRedundantComposable false positive for composable default-argument values

### ISSUE THREAD

**Issue by eboudrant** [slackhq/compose-lints#558]
`ComposeRedundantComposable` only inspects the function body, not parameter default values. Creating false positive for code like : 

```kotlin
@Composable
fun defaultValueFromComposableFunction() = ...

@Composable
fun anyFun(
    colors: Colors = defaultValueFromComposableFunction(),
) {
    // nothing @Composable called here
}
```

Affect versions: `1.5.0`, `1.5.1`

**Comment by kboyarshinov** [comment]
This is also happening for typealiased `@Composable` argument. This test fails:

```kotlin
  @Test
  fun `no errors for composables that take typealiased composable slot`() {
    @Language("kotlin")
    val code =
      """
      import androidx.compose.runtime.Composable
    
      typealias ComposableTypealias = @Composable () -> Unit

      @Composable
      fun Wrapper(content: ComposableTypealias) {
        println("doesn't even call content")
      }
      """
         .trimIndent()
    lint().files(stubs, kotlin(code)).run().expectClean()
  }
```

**Comment by ZacSweers** [comment]
Fixed by #559 and #560
